x-pack/test/api_integration/apis/ml/saved_objects/initialize_trained_models.ts

```diff
@@ -0,0 +1,134 @@
+        ...(jobId ? { metadata: { analytics_config: { id: jobId } } } : {}),
+        input: {
+          field_names: ['common_field'],
+        },
+      } as PutTrainedModelConfig,
+    };
+  }
+
+  describe('GET saved_objects/initialize for trained models', () => {
+    before(async () => {
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/ihp_outlier');
+
+      await spacesService.create({ id: idSpace1, name: 'space_one', disabledFeatures: [] });
+      await spacesService.create({ id: idSpace2, name: 'space_two', disabledFeatures: [] });
+
+      await ml.api.initSavedObjects();
+
+      await ml.api.createDataFrameAnalyticsJob(
+        ml.commonConfig.getDFAIhpOutlierDetectionJobConfig(dfaJobId1),
+        idSpace1
+      );
+      await ml.api.updateJobSpaces(dfaJobId1, 'data-frame-analytics', [idSpace2], [], idSpace1);
+      await ml.api.assertJobSpaces(dfaJobId1, 'data-frame-analytics', [idSpace1, idSpace2]);
+
+      await ml.api.createDataFrameAnalyticsJob(
+        ml.commonConfig.getDFAIhpOutlierDetectionJobConfig(dfaJobIdStarSpace),
+        idSpace1
+      );
+
+      await ml.api.updateJobSpaces(
+        dfaJobIdStarSpace,
+        'data-frame-analytics',
+        ['*'],
+        [idSpace1],
+        idSpace1
+      );
+      await ml.api.assertJobSpaces(dfaJobIdStarSpace, 'data-frame-analytics', ['*']);
+
+      await ml.testResources.setKibanaTimeZoneToUTC();
+    });
+
+    after(async () => {
+      await spacesService.delete(idSpace1);
+      await spacesService.delete(idSpace2);
+      await ml.api.cleanMlIndices();
+      await ml.testResources.cleanMLSavedObjects();
+    });
+
+    it('should initialize trained models and inherit job spaces', async () => {
+      const model = getTestModel(modelIdSpace1, 'classification', dfaJobId1);
+      await ml.api.createTrainedModelES(model.model_id, model.body);
+
+      const body = await runRequest(USER.ML_POWERUSER_ALL_SPACES, 200);
+
+      expect(body).to.have.property('trainedModels');
+      expect(sortBy(body.trainedModels, 'id')).to.eql([{ id: modelIdSpace1 }]);
+      expect(body).to.have.property('success', true);
+      await ml.api.assertTrainedModelSpaces(modelIdSpace1, [idSpace1, idSpace2]);
+    });
+
+    it('should initialize trained models and inherit job star spaces', async () => {
+      const model = getTestModel(modelIdStarSpace, 'classification', dfaJobIdStarSpace);
+      await ml.api.createTrainedModelES(model.model_id, model.body);
+
+      const body = await runRequest(USER.ML_POWERUSER_ALL_SPACES, 200);
+
+      expect(body).to.have.property('trainedModels');
+      expect(sortBy(body.trainedModels, 'id')).to.eql([{ id: modelIdStarSpace }]);
+      expect(body).to.have.property('success', true);
+      await ml.api.assertTrainedModelSpaces(modelIdStarSpace, ['*']);
+    });
+
+    it('should initialize trained models and not inherit spaces', async () => {
+      const model = getTestModel(modelIdSpace2, 'classification');
```

Comment:
Could do with a comment here to explain the effect of not supplying a job ID parameter to `getTestModel`